### PR TITLE
CEP-324: Dynamic Directus Router Revamped (DDRR)

### PR DESCRIPTION
**Pull request by davidpp**
This PR implements the dynamic routing system as described in ADR 001

## Key Changes:
1. Renamed `directusRouteMiddleware` to `directusRouteRouter` for clarity.
2. Implemented direct querying of Directus for page settings translations based on the requested
path.
3. Added support for custom mapping between Directus locales and Next.js locales.
4. Implemented collection-specific settings, particularly for defining custom 'id' field names.
5. Introduced path rewriting to the format `/[mappedLocale]/[collectionName]/[id]`.
6. Added verbose logging in development mode for easier debugging.
7. Moved the router from the directus-next-component package to the directus-next package.
8. Implemented error handling for cases where no translation is found or when a page setting is
not associated with a collection.

## New Features:
- The `directusRouteRouter` function now accepts a configuration object that includes:
  - An optional `localeMap` for custom locale mapping.
  - `collectionSettings` for collection-specific configurations.
- Improved error handling with appropriate logging and 404 responses.
- Development-mode logging for better visibility into the routing process.

## Usage:
Projects using this stack will need to update their `middleware.ts` files to include the necessary
configuration and matcher. Example is visible in this PR: https://github.com/OKAMca/cepsum-mono/pull/157

## Testing
1. In Stack, publish the routing package using `npx nx run @stack-ui/source:publish-local --libraries=directus-next`
2. In Cepsum, install package.  Replace timestamp from command outputted by Step 1. with @latest at the end instead.
- Ask David for now.

##Demo
https://cln.sh/sp0T0JGS

## Summary by CodeRabbit

- **New Features**
	- Introduced a routing mechanism for handling requests in Directus applications using Next.js.
	- Added support for server-side rendering capabilities in the Directus Next library.

- **Improvements**
	- Expanded build configuration to include multiple entry points and enhanced modularity.
	- Updated library branding from `directus-block` to `directus-next`.
	- Enhanced modularization with the addition of a server module export.

- **Bug Fixes**
	- Implemented error handling for missing GraphQL configuration during request processing.

### CODE REVIEW

**Comment by davidpp** [comment]
@marie-maxime @yanmorinokamca je peux avoir votre avis sur https://github.com/OKAMca/stack/blob/refactor-directus-rewrite/docs/ADR/001_directus-router.md

Est-ce que j'ai oublié quelque chose d'important? je peux vous expliquer de vive voix au besoin.

File: package.json [hunk 1]
```diff
@@ -87,6 +86,7 @@
     "@commitlint/config-conventional": "18.4.3",
     "@graphql-eslint/eslint-plugin": "^3.20.1",
     "@nx/cypress": "19.3.1",
+    "@nx/devkit": "^19.6.2",
```
Comment (marie-maxime) [inline package.json:89]: les packages de next devrait toujours être la meme version pour prévenir des bugs entre les packages et facilité le processus d'upgrade de nx

**Review by marie-maxime — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -44,23 +44,64 @@ Targets can be defined in the `package.json` or `projects.json`. Learn more [in
 
 Have a look at the [Nx Console extensions](https://nx.dev/nx-console). It provides autocomplete support, a UI for exploring and running tasks & generators, and more! Available for VSCode, IntelliJ and comes with a LSP for Vim users.
 
-## Ready to deploy?
 
-Just run `nx build demoapp` to build the application. The build artifacts will be stored in the `dist/` directory, ready to be deployed.
+Certainly! Here's an updated section for your README that includes the requirement to start the local registry before running the `publish-local` command.
```
[the remark this answers is not in the document]
Comment (davidpp) [reply README.md:48]: oopsie.

 done dans l'autre PR parcontre.

**Comment by davidpp** [comment]
Closing for new PR without local dev mode

**Pull request closed without merging**